src/wagtail_ai/static_src/image_description/main.tsx

```diff
@@ -0,0 +1,49 @@
+import './main.css';
+import { fetchResponse } from '../api';
+
+import wandIcon from '!!raw-loader!../wand_icon.svg';
+
+document.addEventListener('wagtail-ai:image-form', (event) => {
+  const input = event.target as HTMLInputElement;
+  const imageId = input.dataset['wagtailai-image-id']!;
+  const csrfToken = (
+    input.form!.querySelector(
+      'input[name=csrfmiddlewaretoken]',
+    ) as HTMLInputElement
+  ).value;
+
+  const flexWrapper = document.createElement('div');
+  flexWrapper.classList.add('wagtailai-input-wrapper');
+  input.parentNode!.replaceChild(flexWrapper, input);
+  flexWrapper.appendChild(input);
+
+  const button = document.createElement('button');
+  button.type = 'button';
+  button.title = input.dataset['wagtailai-button-title'] || '';
+  button.classList.add('wagtailai-button');
+  button.innerHTML = wandIcon;
+  flexWrapper.appendChild(button);
+
+  let errorMessage: HTMLElement | null = null;
+
+  button.addEventListener('click', async () => {
+    errorMessage?.parentNode?.removeChild(errorMessage);
+    button.disabled = true;
+    button.classList.add('loading');
+
+    try {
+      const formData = new FormData();
+      formData.append('image_id', imageId);
+      formData.append('csrfmiddlewaretoken', csrfToken);
+      input.value = await fetchResponse('DESCRIBE_IMAGE', formData);
```

Comment:
**Question:** Would there be an option for custom alt-text fields this way or are we limited by Wagtail in only being allowed to use the title field?

Reply:
I'm not sure if this is what you're asking, but the JS code will happily work with any text input field, if your custom image model has such extra fields. It's briefly mentioned in the docs: https://github.com/wagtail/wagtail-ai/pull/81/files#diff-da6a4f33a37dc72141b415345529ef7c066f01a4413477e569f89ddd51d39276R46.